experiment/template/eVOLVER.py

```diff
@@ -482,6 +482,52 @@ def calc_growth_rate(self, vial, gr_start, elapsed_time):
         text_file.write("{0},{1}\n".format(elapsed_time, slope))
         text_file.close()
 
+    def tail_to_np(self, path, window=10):
+        f = open(path, 'rb')
+        if window == 0:
+            return []
+
+        BUFSIZ = 1024
+        f.seek(0, 2)
+        remaining_bytes = f.tell()
+        size = window + 1
+        block = -1
+        data = []
+
+        while size > 0 and remaining_bytes > 0:
+            if remaining_bytes - BUFSIZ > 0:
+                # Seek back one whole BUFSIZ
+                f.seek(block * BUFSIZ, 2)
+                # read BUFFER
+                bunch = f.read(BUFSIZ)
+            else:
+                # file too small, start from beginning
+                f.seek(0, 0)
+                # only read what was not read
+                bunch = f.read(remaining_bytes)
+
+            bunch = bunch.decode('utf-8')
+            data.insert(0, bunch)
```

Comment:
Maybe comment why you insert at 0 (essentially reversing order). It's also possible to append instead of insert and just reverse at the end. I think that is slightly more efficient, although it doesn't really matter as we are only reading a few lines.